### PR TITLE
Correct regexp terminology

### LINKED ISSUE #3337: Add guide for RegExp match indices
## What is the new suggestion?
In https://github.com/mdn/content/pull/3284 I've added the descriptions of the regular expression match indices. What's still missing, though, is a guide page for this feature.

## Why is it important or useful?
This should be a subpage of the [existing regular expressions guide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions).

The page should explain in more detail why match indices are useful and are created and used including some examples.

Sebastian

MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions

### LINKED ISSUE #15227: RegExp terminology
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges

### What information was incorrect, unhelpful, or incomplete?

RegExp terminology can be confusing, let's not confuse the reader's minds further...

The section title refers to groups and ranges, but goes on to include disjunctions, character classes and back references.

`a|b` is not a group it is a disjunction. `[abc]` is not a range, it is a character class. `\1` and `\k<name>` are neither.

Most of the constructs described in this page are atoms, per the spec, with the exception of `a|b`

The JS API complicates the matter by using `result.groups` to refer to the values of named capturing groups (anonymous capturing groups and non-capturing groups are also groups, but they are excluded from that badly named object).

### What did you expect to see?

Use consistent terminology. Consistent with the spec, and as consistent as possible with itself (`results.group` has sailed, we won't put it back in the box).

I can rewrite some of the page, but I'd like to have the get go before I invest too much time for nothing.

This page could be limited to atoms, and we could create one for sequences (what the spec calls an alternative) and disjunctions, which are not atomic.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/guide/regular_expressions/groups_and_ranges`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/guide/regular_expressions/groups_and_ranges/index.md
* Last commit: https://github.com/mdn/content/commit/303a6c3b8ec6658d2956c191c50aee0a0e9e4da5
* Document last modified: 2022-02-18T09:30:25.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Summary

Fix #15227
Fix #3337 (through the addition of one example; I hope this suffices, because it's TBH not a hugely useful feature for everyone.)

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/guide/regular_expressions/character_classes/index.md [hunk 1]
```diff
@@ -251,6 +307,23 @@ Character classes distinguish kinds of characters such as, for example, distingu
...
+      </td>
+      <td>
+        <p>
+          <strong>Disjunction: </strong>Matches either "x" or "y". Each component, separated by a pipe (<code>|</code>), is called an <em>alternative</em>. For example,
+          <code>/green|red/</code> matches "green" in "green apple" and "red" in
+          "red apple".
+        </p>
+        <div class="notecard note">
+          <p>
+            <strong>Note:</strong> A disjunction is another way to specify "a set of choices", but it's not a character class. Disjunctions are not atoms — you need to use a <a href="/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Backreferences">group</a> to make it part of a bigger pattern. <code>[abc]</code> is functionally equivalent to <code>(?:a|b|c)</code>.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/guide/regular_expressions/character_classes/index.md:322]: I know it looks weird to say "we are on the character class page but this is not a character class"... But I don't know where else to put this (it's certainly less like a group than a character class), and I don't want to create another page just for disjunctions.

File: files/en-us/web/javascript/guide/regular_expressions/character_classes/index.md [hunk 2]
```diff
@@ -26,6 +26,62 @@ Character classes distinguish kinds of characters such as, for example, distingu
...
   <tbody>
+    <tr>
+      <td>
+        <code>[xyz]<br />[a-c]</code>
+      </td>
+      <td>
+        <p>
+          A character class. Matches any one of the enclosed characters. You can
+          specify a range of characters by using a hyphen, but if the hyphen
+          appears as the first or last character enclosed in the square brackets
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/guide/regular_expressions/character_classes/index.md:37]:
```suggestion
          appears as the first or last character enclosed in the square brackets,
```

File: files/en-us/web/javascript/guide/regular_expressions/character_classes/index.md [hunk 3]
```diff
@@ -26,6 +26,62 @@ Character classes distinguish kinds of characters such as, for example, distingu
...
+        <p>
+          <code>[^xyz]<br />[^a-c]</code>
+        </p>
+      </td>
+      <td>
+        <p>
+          A negated or complemented character class. That is, it matches
+          anything that is not enclosed in the brackets. You can specify a range
+          of characters by using a hyphen, but if the hyphen appears as the
+          first or last character enclosed in the square brackets it is taken as
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/guide/regular_expressions/character_classes/index.md:68]:
```suggestion
          first or last character enclosed in the square brackets, it is taken as
```

File: files/en-us/web/javascript/guide/regular_expressions/cheatsheet/index.md [hunk 4]
```diff
@@ -23,6 +23,63 @@ This page provides an overall cheat sheet of all the capabilities of `RegExp` sy
...
   <tbody>
+    <tr>
+      <td>
+        <code>[xyz]<br />[a-c]</code>
+      </td>
+      <td>
+        <p>
+          A character class. Matches any one of the enclosed characters. You can
+          specify a range of characters by using a hyphen, but if the hyphen
+          appears as the first or last character enclosed in the square brackets
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/guide/regular_expressions/cheatsheet/index.md:34]:
```suggestion
          appears as the first or last character enclosed in the square brackets,
```

File: files/en-us/web/javascript/guide/regular_expressions/cheatsheet/index.md [hunk 5]
```diff
@@ -23,6 +23,63 @@ This page provides an overall cheat sheet of all the capabilities of `RegExp` sy
...
+        <p>
+          <code>[^xyz]<br />[^a-c]</code>
+        </p>
+      </td>
+      <td>
+        <p>
+          A negated or complemented character class. That is, it matches
+          anything that is not enclosed in the brackets. You can specify a range
+          of characters by using a hyphen, but if the hyphen appears as the
+          first or last character enclosed in the square brackets it is taken as
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/guide/regular_expressions/cheatsheet/index.md:66]:
```suggestion
          first or last character enclosed in the square brackets, it is taken as
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**